pkg/proc/eval.go

```diff
@@ -3044,49 +3182,88 @@ func (v *Variable) findMethod(mname string) (*Variable, error) {
 
 		//TODO(aarzilli): support generic functions?
 
-		if fns := v.bi.LookupFunc()[fmt.Sprintf("%s.%s.%s", pkg, receiver, mname)]; len(fns) == 1 {
+		rv, err := lookupMethod(v, isptr, pkg, receiver, mname)
+		if err != nil {
+			return nil, err
+		} else if rv != nil {
+			return rv, nil
+		}
+
+		// queue embedded fields for search
+		structVar := v.maybeDereference()
+		structVar.Name = v.Name
+		if structVar.Unreadable != nil {
+			return structVar, nil
+		}
+		switch t := structVar.RealType.(type) {
+		case *godwarf.StructType:
+			for _, field := range t.Field {
+				if field.Embedded {
+					embeddedVar, err := structVar.toField(field)
+					if err != nil {
+						return nil, err
+					}
+					queue = append(queue, embeddedVar)
+				}
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+func lookupMethod(v *Variable, isptr bool, pkg, receiver, name string) (*Variable, error) {
+	if isptr {
```

Comment:
Can you explain this check? IIRC go won't allow the same method name on pointer and non-pointer receiver, it shouldn't matter which order you look them up.